futures-util/src/stream/poll_immediate.rs

```diff
@@ -0,0 +1,80 @@
+use futures_core::task::{Context, Poll};
+use futures_core::Stream;
+use std::pin::Pin;
+
+/// Stream for the [`poll_immediate`](poll_immediate()) function.
+#[derive(Debug, Clone)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct PollImmediate<S>(Option<S>);
+
+impl<T, S> Stream for PollImmediate<S>
+where
+  S: Stream<Item = T>,
+{
+    type Item = Poll<T>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        unsafe {
+            // # Safety
+            // We never move the inner value until it is done. We only get a reference to it.
+            let inner = &mut self.get_unchecked_mut().0;
+            let fut = match inner.as_mut() {
+                // inner is gone, so we can continue to signal that the stream is closed.
+                None => return Poll::Ready(None),
+                Some(inner) => inner,
+            };
+            let stream = Pin::new_unchecked(fut);
+            match stream.poll_next(cx) {
+                Poll::Ready(Some(t)) => Poll::Ready(Some(Poll::Ready(t))),
+                Poll::Ready(None) => {
+                    // # Safety
+                    // The inner stream is done, so we need to drop it. We do it without moving it
+                    // by using drop in place. We then write over the value without trying to drop it first
+                    // This should uphold all the safety requirements of `Pin`
+                    std::ptr::drop_in_place(inner);
+                    std::ptr::write(inner, None);
```

Comment:
same here

Reply:
I found the set() function in the abortable code.